src/placeos-driver/proxy/system.cr

```diff
@@ -4,20 +4,42 @@ require "../driver_model"
 require "./remote_driver"
 
 class PlaceOS::Driver::Proxy::System
+  # Local system available to logic driver
   def initialize(
-    @model : DriverModel::ControlSystem,
+    model : DriverModel::ControlSystem,
     @reply_id : String,
     @logger : ::Log = ::Log.for("driver.proxy.system"),
     @subscriptions : Proxy::Subscriptions = Proxy::Subscriptions.new
   )
-    @system_id = @model.id
+    @config = model
+    @system_id = model.id
     @system = PlaceOS::Driver::RedisStorage.new(@system_id, "system")
   end
 
+  # Remote system (lazily loaded config)
+  def initialize(
+    @system_id : String,
+    @reply_id : String,
+    @logger : ::Log = ::Log.for("driver.proxy.system"),
+    @subscriptions : Proxy::Subscriptions = Proxy::Subscriptions.new
+  )
+    @system = PlaceOS::Driver::RedisStorage.new(@system_id, "system")
```

Comment:
Is it worth validating existence of `system_id` here? I can't think of a situation where this would load before that exists, but can think of many where an incorrect one may be piped in, but not read from until some point in the future (causing an error then).

Reply:
it's quite possible that a system comes online before another one
or a system is deleted after initialization
or a system has no drivers in it (resulting in an empty hash)

the redis model is basically empty until it has some data - so kind of hard to tell if a system exists without fetching the model - which is overhead we want to avoid...

probably best to check as part of the driver code using remote systems `system("1234").exists?(:Display_1)`
and expose some debugging state for simplified fault correction

Reply:
Could pipe those ID's into a bloom filter as systems are created / load, then pop that filter onto redis. IIRC there's a bloom filter module, but would also be simple enough to slam it straight onto a bitmap to keep things as standard as possible from env requirements.